tensorboard/plugins/hparams/backend_context.py

```diff
@@ -332,16 +332,20 @@ def _compute_hparam_info_from_values(self, name, values):
                 if _can_be_converted_to_string(v)
             )
             result.domain_discrete.extend(distinct_string_values)
+            result.differs = len(distinct_string_values) > 1
 
         if result.type == api_pb2.DATA_TYPE_BOOL:
-            result.domain_discrete.extend([True, False])
+            distinct_bool_values = set(v.bool_value for v in values)
```

Comment:
I would prefer to keep the domain to be the constant [True, False]. But I agree that we should calculate differs based on the actual values.

Reply:
Done. Does this apply to hparams from the new internal source as well?

Reply:
Huh, ya, now that you remind me to think about how we do this for internal sources, I think you may have been correct the first time (calculate the boolean domains based on value). Sorry for the churn!

Reply:
Nw! If you prefer to set them directly we can make that change too.